WIP: paymail avatars

ryanxcharles opened the pull request.

ryanxcharles committed:
contact cards

ryanxcharles committed:
minor updates

ryanxcharles committed:
typo

ryanxcharles committed:
"enforce" square images

rt121212121 commented on src/07-contact-cards.md:
```diff
@@ -0,0 +1,69 @@
+
+The returned image _MUST_ be a square, at the resolution specified by `s`, and should look good when cropped to a circle, since this will be the most common way to display the images.
+
+Note, however, that there is no way to enforce what resolution image is returned or what the viewer will do with an image that is the incorrect resolution. Thus, if the returned image is not a square or if it is the wrong resolution, the viewer behavior is undefined. Some viewers may crop the image, some may squish the image, and some may choose not to display the image at all.
```
I wonder if this last paragraph waters down the standard of what a standards compliant avatar service is and undermines it somewhat. It effectively suggests that application developers have to just deal with non-standards compliant avatar services, and there is no standard just a hope that services will do what is suggested.

What if we proscribed what a standards compliant application developer should do - they still are not required to do it, nor are services required to meet the suggested behaviour, but by doing so it pushes the avatar ecosystem into compliance from the beginning and more or less ensures the quality of how they appear in all our applications by doing so.

1. Avatar urls should only ever return an image of the requested size, or if no size is provided, 80x80.
2. Avatar urls should only ever return images in the JPEG, PNG and GIF formats.
3. If an avatar url returns a non-square image, or a format not listed above, applications should consider the avatar service broken and should not use and should stop requesting their avatars.

ryanxcharles replied:
From the perspective of the web, it is easier to just resize the image than it is to try to detect the size of the image and display something different instead. Thus from our POV we'd rather not have to be bothered with detecting it in the UI. I'm not sure about other languages/platforms.

So here are some options:
- Invalidly sized images are always squished to a square (easiest option for web apps).
- Invalidly sized images are always cropped (second easiest option for web apps).
- Invalidly sized images are detected and not displayed (hardest option for web apps).

The last option does indeed seem to be the most likely to force compliance of square images.

ryanxcharles replied:
I will update the PR with your suggestion of more strongly worded language.

ryanxcharles committed:
rename "contact card" -> "public profile"

Zheming Lin pointed out there is already a standard for contact cards
called "vcard". However, this standard is actually not about the private
information that is usually shared inside of a contact card. It is for
the public profile information that anyone can access with no
authentication. It is specifically for name & avatar, and possibly other
public information like website. It is not for personal shipping address
or anything like that. Thus I am renaming it to "public profile" to
reflect the proper nature of the standard.

ryanxcharles committed:
following suggestion of RT, use stronger language

This version enforces square images and the correct resolution. This is
part of the value of a standard.

ryanxcharles committed:
add hash

ryanxcharles committed:
update default size to 180x180
